drivers/xen/9pfront/Makefile.uk

```diff
@@ -0,0 +1,8 @@
+$(eval $(call addlib_s,lib9pfront,$(CONFIG_LIB9PFRONT)))
+
+# FIXME: Extra includes should be removed
```

Comment:
That libraries should not just include some foreign headers by themselves if there is no good reason. This is a leftover that needs fixing with rearch.

Reply:
Okay, I guess you referring to the direct inclusions of xenplat / and xenbus. Then maybe we can replace "extra headers" with a more clear explanation based on what you posted above? Or separate these two `CINCLUDES` from others so that it's more clear that the comment refers to these? On some drivers that depend on `plat ` we have marked such dependencies similarly to:
```
...
LIBUKBUS_PCI_CINCLUDES-y += -I$(LIBUKBUS_PCI_BASE)/include

# TODO Remove as soon as plat dependencies go away
LIBUKBUS_PCI_CINCLUDES-y += -I$(UK_PLAT_COMMON_BASE)/include

LIBUKBUS_PCI_SRCS-y += $(LIBUKBUS_PCI_BASE)/pci_bus.c
...
```

Reply:
Yes, correct: Will do that :+1:

Reply:
Yes, exactly. I tried a bit and found finally a solution that removes these lines already from the Xen drivers. There are two more commits on this PR that I'd like you to check. But basically I achieved it by the following:
- Properly define and export the public API of `drivers/xen/xenbus`
- Similar to other libraries, export headers and compile flags of `libxenplat` globally: I basically define an API.

Reply:
Nice 👌🏼 I like this intermediate step approach, as it allows at least drivers to be clean.

>    With this change, the compile flag `CONFIG_PARAVIRT` is name spaced to  `XEN_PARAVIRT`, as well.

Could you elaborate on the commit message with some reasoning for that?

>  Because of this and the fact that I try to keep my commits "bisect"-able, I did not change the message because the latest commits will remove it again.

I'm all on for bisectability, but IMO that is not a reason to introduce transient changes. The commits in a PR should reflect the final state of the work, and not the development process. So to go by the book, the API should had been defined in an earlier commit. Anyway, I know restructuring commits is a mundane task, so if you are satisfied with the current state, I won't stand on the way.

Reply:
> Could you elaborate on the commit message with some reasoning for that change?
I updated the commit message to point out that with the platform rearch we are basically removing the platform-internal scope where drivers were originally in. This means that API definitions are public usable, even by application code. Because of this, proper name spacing is required.

Reply:
> I'm all on for bisectability, but IMO that is not a reason to introduce transient changes. The commits in a PR should reflect the final state of the work, and not the development process. So to go by the book, the API should had been defined in an earlier commit. Anyway, I know restructuring commits is a mundane task, so if you are satisfied with the current state, I won't stand on the way.

If you mean the exporting of the XenBus API, the commit is actually doing only that. The API is not redefined or changed, just "properly" exported by moving files. But I agree with your general statement that irrespective of bisect-able commits or not: transition commits do not make much sense and should be avoided. Every commit should directly go towards the final target. With a proper commit history, this is typically feasible.